bandcampmobile/content/contents/code/bandcampm.js

```diff
@@ -0,0 +1,119 @@
+/* http://bandcamp.com resolver for Tomahawk.
+ * This one use API used by Bandcamp's Android App
+ * Written in 2015 by Anton Romanov
+ * Licensed under the Eiffel Forum License 2.
+ */
+
+var BandcampResolver = Tomahawk.extend( Tomahawk.Resolver, {
+    apiVersion: 0.9,
+
+    settings: {
+        cacheTime: 300,
+        name: 'bandcampm',
+        weight: 90,
+        icon: '../images/icon.png',
+        timeout: 15
+    },
+
+    _convertTrack: function (entry) {
+        return {
+            artist:     entry.band_name,
+            album:      entry.album_name,
+            track:      entry.name,
+            title:      entry.name,
+            hint:        'bandcampm://track/' + entry.band_id + '/' + entry.id,
+            bitrate:    128,
+            type:       "track"
+        };
+    },
+
+    _getEditDistance : function(a, b){
```

Comment:
Theres a comment about that in the code:
There are no mp3 urls in the returned search result, even more - there is no flag indicating you can get the file. You have to query every track id individually to find that out.
With number of results you get from the service if you start making requests for each entry to find if its streamable - you start being throttled almost immediattely. The idea is to only check copule of the most relevant results